ui/app/services/path-help.js

```diff
@@ -42,89 +44,130 @@ export default Service.extend({
-        } else {
-          //we need the mount config
-          path = paths.configPath[0].path;
+        paths = itemType ? this.filterPathsByItemType(pathInfo, itemType) : pathInfo.paths;
+        const createPath = paths.find(path => path.operations.includes('post') && path.action !== 'Delete');
+        path = createPath.path;
+        path = path.includes('{') ? path.slice(0, path.indexOf('{') - 1) + '/example' : path;
+        if (!path) {
+          return reject();
         }
-        helpUrl = `/v1/${apiPath}${path.slice(1)}?help=true`;
+
+        helpUrl = `/v1/${apiPath}${path.slice(1)}?help=true` || newModel.proto().getHelpUrl(backend);
+        pathInfo.paths = paths;
+        newModel = newModel.extend({ paths: pathInfo });
         return this.registerNewModelWithProps(helpUrl, backend, newModel, modelName);
+      })
+      .catch(err => {
+        console.error(err);
       });
-    }
   },
 
-  reducePaths(paths, currentPath) {
+  reducePathsByPathName(pathInfo, currentPath) {
     const pathName = currentPath[0];
-    const pathInfo = currentPath[1];
+    const pathDetails = currentPath[1];
+    const displayAttrs = pathDetails['x-vault-displayAttrs'];
 
-    //config is a get/post endpoint that doesn't take route params
-    //and isn't also a list endpoint and has an Action of Configure
-    if (
-      pathInfo.post &&
-      pathInfo.get &&
-      (pathInfo['x-vault-displayAttrs'] && pathInfo['x-vault-displayAttrs'].action === 'Configure')
-    ) {
-      paths.configPath.push({ path: pathName });
-      return paths; //config path should only be config path
+    if (!displayAttrs) {
+      return pathInfo;
     }
 
-    //list endpoints all have { name: "list" } in their get parameters
-    if (pathInfo.get && pathInfo.get.parameters && pathInfo.get.parameters[0].name === 'list') {
-      paths.list.push({ path: pathName });
+    if (pathName.includes('{')) {
+      //we need to know if there are url params
+      pathName.split('{')[1].split('}')[0];
     }
 
-    if (pathInfo.delete) {
-      paths.delete.push({ path: pathName });
+    let itemType, itemName;
+    if (displayAttrs.itemType) {
+      itemType = displayAttrs.itemType;
+      let items = itemType.split(':');
+      itemName = items[items.length - 1];
+      items = items.map(item => dasherize(singularize(item.toLowerCase())));
+      itemType = items.join('~*');
     }
 
-    //create endpoints have path an action (e.g. "Create" or "Generate")
-    if (pathInfo.post && pathInfo['x-vault-displayAttrs'] && pathInfo['x-vault-displayAttrs'].action) {
-      paths.create.push({
-        path: pathName,
-        action: pathInfo['x-vault-displayAttrs'].action,
-      });
+    if (itemType && !pathInfo.itemTypes.includes(itemType)) {
+      pathInfo.itemTypes.push(itemType);
     }
 
-    if (pathInfo['x-vault-displayAttrs'] && pathInfo['x-vault-displayAttrs'].navigation) {
-      paths.navPaths.push({ path: pathName });
+    let operations = [];
```

Comment:
`const` vs `let` nit

Reply:
i've left this one in since we're adding to the array right after. these nits a great though!

Reply:
Hey!

`const` vs `let` is about immutable binding not immutable values, this my favourite article on it which explains a bit further:

https://mathiasbynens.be/notes/es6-const

So yes, you push values onto the array and change the array, but the binding of the array to `operations` never changes, `operations` is never re-bound.

Still very nitty though! Any completely up to you if you wanna change. But I think as we move forwards and we say goodbye to IE and use less babelscript this is going to be vaguely more important.

Reply:
ah, yeah that all makes sense. thank you so much for your explanation + the added resource! this is all very helpful. fixed! :D